dlp/snippets/inspect/inspect_with_custom_regex.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package inspect
+
+// [START dlp_inspect_custom_regex]
+import (
+	"context"
+	"fmt"
+	"io"
+
+	dlp "cloud.google.com/go/dlp/apiv2"
+	"cloud.google.com/go/dlp/apiv2/dlppb"
+)
+
+// inspectWithCustomRegex inspect a data with custom regex pattern
+func inspectWithCustomRegex(w io.Writer, projectID, textToInspect, customRegexPattern, infoTypeName string) error {
+	//projectID := "my-project-id"
+	//textToInspect := "Patients MRN 444-5-22222"
+	//customRegexPattern := "[1-9]{3}-[1-9]{1}-[1-9]{5}"
+	//infoTypeName := "C_MRN"
+
+	ctx := context.Background()
+
+	// Initialize a client once and reuse it to send multiple requests. Clients
+	// are safe to use across goroutines. When the client is no longer needed,
+	// call the Close method to cleanup its resources.
+	client, err := dlp.NewClient(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Closing the client safely cleans up background resources.
+	defer client.Close()
+
+	// Specify the type and content to be inspected.
+	var contentItem = &dlppb.ContentItem{
+		DataItem: &dlppb.ContentItem_ByteItem{
+			ByteItem: &dlppb.ByteContentItem{
+				Type: dlppb.ByteContentItem_TEXT_UTF8,
+				Data: []byte(textToInspect),
+			},
+		},
+	}
+
+	// Construct the custom regex detectors
+	var customInfoType = &dlppb.CustomInfoType{
+		InfoType: &dlppb.InfoType{
+			Name: infoTypeName,
+		},
+		// Specify the regex pattern the inspection will look for.
+		Type: &dlppb.CustomInfoType_Regex_{
+			Regex: &dlppb.CustomInfoType_Regex{
+				Pattern: customRegexPattern,
+			},
+		},
+		Likelihood: dlppb.Likelihood_POSSIBLE,
+	}
+
+	// Construct the Inspect request to be sent by the client.
+	req := &dlppb.InspectContentRequest{
+		Parent: fmt.Sprintf("projects/%s/locations/global", projectID),
+		Item:   contentItem,
+		// Construct the configuration for the Inspect request.
+		InspectConfig: &dlppb.InspectConfig{
+			CustomInfoTypes: []*dlppb.CustomInfoType{
+				customInfoType,
+			},
+			IncludeQuote: true,
+		},
+	}
+
+	// Send the request.
+	resp, err := client.InspectContent(ctx, req)
+	if err != nil {
+		fmt.Fprintf(w, "Receive: %v", err)
```

Comment:
issue: just return the error as-is to the caller.

Reply:
fixed it.